src/constants/content.ts

```diff
@@ -1,20 +1,29 @@
-import DATA, { BookID, BookType } from '@data/contrarianmba.json';
+import DATA, { BookID } from '@data/contrarianmba.json';
 
 import { Book, Category } from '@/types';
 
 export const CATEGORIES: Category[] = DATA.categories.map(
     (category) => new Category(category)
 );
 
-export const CATEGORIES_BY_SLUG: { [key: string]: Category } = {};
-CATEGORIES.forEach((category) => {
-    CATEGORIES_BY_SLUG[category.slug] = category;
-});
+export const CATEGORIES_BY_SLUG: { [key: string]: Category } =
+    CATEGORIES.reduce(
+        (collection, category) => ({
+            ...collection,
+            [category.slug]: category,
+        }),
+        {}
+    );
```

Comment:
# Reviewer Note
Idiomatic way of JavaScript.

Reply:
I'm not sure, I feel like `...collection` will actually make this an `O(n^2)` operation, instead of `O(n)`